docs/expression_syntax.md

```diff
@@ -0,0 +1,195 @@
+**Is your feature request related to a problem? Please describe.**
+As part of a larger feature (#522) to support complex condition statements in data prepper there is a need to define a syntax for conditional statements. A conditional statement is a String that is evaluated at runtime and may reference fields within a record.
+
+**Describe the solution you'd like**
+_Terms used throught this document are defined in the Definitions section_
+
+## Supported Operators
+In order of evaluation priority. _(top to bottom, left to right)_
+
+| Operator             | Description              | Data Prepper Version |
+|----------------------|--------------------------|----------------------|
+| `{}`                 | Set Initializer          | 1.4.0                |
+| `()`                 | Priority Expression      | 1.3.0                |
+| `not`                | Not Operator             | 1.3.0                |
+| `in`, `not in`       | Set Operators            | 1.4.0                |
+| `<`, `<=`, `>`, `>=` | Relational Operators     | 1.3.0                |
+| `=~`, `!~`           | Regex Equality Operators | 2.0.0                |
+| `==`, `!=`           | Equality Operators       | 1.3.0                |
+| `and`, `or`          | Conditional Expression   | 1.3.0                |
+| `,`                  | Set Value Delimiter      | 1.4.0                |
+
+## Reserved for possible future functionality
+Reserved symbol set: `^`, `*`, `/`, `%`, `+`, `-`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, `${<text>}`
+
+## Set Initialiser
+Defines a set or term and/or expressions.
+
+Examples
+```
+{1, 2, 3}
```

Comment:
For the documentation, I'd like to use more realistic examples. e.g. comparing against status code.

Reply:
A better solution could be to remove the sections referring to operators not yet supported in 1.3.0